test/Microsoft.Health.Fhir.Tests.E2E/Common/FhirClient.cs

```diff
@@ -268,44 +268,52 @@ private async Task<FhirResponse<T>> CreateResponseAsync<T>(HttpResponseMessage r
                 string.IsNullOrWhiteSpace(content) ? null : (T)_deserialize(content));
         }
 
-        private async Task SetupAuthenticationAsync(TestApplication clientApplication, TestUser user = null)
+        private async Task SetupAuthenticationAsync(TestApplication clientApplication, TestUser user = null, AuthenticationScenarios authenticationScenarios = AuthenticationScenarios.VALIDAUTH)
         {
             await GetSecuritySettings("metadata");
 
             if (SecuritySettings.SecurityEnabled)
             {
                 var tokenKey = $"{clientApplication.ClientId}:{(user == null ? string.Empty : user.UserId)}";
+
                 if (!_bearerTokens.TryGetValue(tokenKey, out string bearerToken))
                 {
-                    bearerToken = await GetBearerToken(clientApplication, user);
+                    bearerToken = await GetBearerToken(clientApplication, user, authenticationScenarios);
                     _bearerTokens[tokenKey] = bearerToken;
                 }
 
                 HttpClient.SetBearerToken(bearerToken);
             }
         }
 
-        private async Task<string> GetBearerToken(TestApplication clientApplication, TestUser user)
+        private async Task<string> GetBearerToken(TestApplication clientApplication, TestUser user, AuthenticationScenarios authenticationScenarios)
         {
-            var formContent = new FormUrlEncodedContent(user == null ? GetAppSecuritySettings(clientApplication) : GetUserSecuritySettings(clientApplication, user));
+            var formContent = new FormUrlEncodedContent(user == null ? GetAppSecuritySettings(clientApplication, authenticationScenarios) : GetUserSecuritySettings(clientApplication, user));
 
             HttpResponseMessage tokenResponse = await HttpClient.PostAsync(SecuritySettings.TokenUrl, formContent);
 
             var tokenJson = JObject.Parse(await tokenResponse.Content.ReadAsStringAsync());
 
-            var bearerToken = tokenJson["access_token"].Value<string>();
-            return bearerToken;
+            if (tokenJson["access_token"] != null)
+            {
+                return tokenJson["access_token"].Value<string>();
+            }
+
+            return null;
```

Comment:
If we get an intermittent failure here, it will set the dictionaries value to null and should fail as a 401... This could end up passing certain tests and failing others not as expected. Would it be better to handle the case we're trying to handle explicitly via the authenticationscenario? (e.g. just eagerly return null when we don't want a token)

Reply:
This could still happen if we didn't get an `access_token` in a response due to an external issue when we expect to get a token.